game-core/src/main/java/games/strategy/engine/framework/startup/ui/panels/main/game/selector/GameFileSelector.java

```diff
@@ -32,15 +36,26 @@ private GameFileSelector() {}
       fileDialog.setVisible(true);
       final String fileName = fileDialog.getFile();
       final String dirName = fileDialog.getDirectory();
-      return Optional.ofNullable(fileName).map(name -> new File(dirName, fileName));
+      return Optional.ofNullable(fileName)
+          .map(name -> new File(dirName, fileName))
+          .map(this::mapFileResult);
     }
 
     // Non-Mac platforms should use the normal Swing JFileChooser
     final JFileChooser fileChooser = SaveGameFileChooser.getInstance();
     final int selectedOption = fileChooser.showOpenDialog(owner);
     if (selectedOption == JFileChooser.APPROVE_OPTION) {
-      return Optional.of(fileChooser.getSelectedFile());
+      return Optional.of(fileChooser.getSelectedFile()).map(this::mapFileResult);
     }
     return Optional.empty();
   }
+
+  private File mapFileResult(final File file) {
+    if (file.exists()) {
+      return file;
+    } else {
+      fileDoesNotExistAction.accept(file);
```

Comment:
It's a bit hard to wrap my head around this, this is a super weird side-effect, but ok.
Perhaps worth annotating this method with `@Nullable`

Reply:
Well from just looking at the other method I assumed that `mapFileResult` was a "pure" function with no side-effects, so I was quite suprised to see the error handler there.
Maybe not "weird" but definitely unexpected.